### PR TITLE
Remove skins, make new search UI work in Plone 4 & fix all doctests

### PR DESCRIPTION
**Pull request by tomgross**
This PR
- removes skins folder from product
- makes react based search work with Plone 4
- Allows to edit boost script in the control panel
- Remove search profile -> customized search is now available always with product install

### CODE REVIEW

**Comment by tomgross** [comment]
@tisto can you have a look on the failing tests please. They are caused by the switch to the new search view and you know it better :wink:

**Comment by tisto** [comment]
@tomgross can we try to go in small (or even tiny) steps? Otherwise it is hard to tell why those tests fail. Bottom line is that we have to replace all the doctests with robot tests.

**Comment by tisto** [comment]
@tomgross can you have a look at the failing Plone 5 build? It seems zcml:condition does not work as expected.

```
<InterfaceClass zope.publisher.interfaces.browser.IBrowserRequest>, <InterfaceClass collective.solr.browser.interfaces.IThemeSpecific>)
    File "/home/travis/build/collective/collective.solr/src/collective/solr/browser/configure.zcml", line 65.2-73.8
        <browser:page
            name="search"
            class="Products.CMFPlone.browser.search.Search"
            permission="zope2.View"
            for="plone.app.layout.navigation.interfaces.INavigationRoot"
            template="search.pt"
            layer=".interfaces.IThemeSpecific"
            zcml:condition="have plone-5"
            />
    File "/home/travis/build/collective/collective.solr/src/collective/solr/browser/configure.zcml", line 75.2-83.8
        <browser:page
            name="search"
            class=".p4_search.Search"
            permission="zope2.View"
            for="plone.app.layout.navigation.interfaces.INavigationRoot"
            template="search.pt"
            layer=".interfaces.IThemeSpecific"
            zcml:condition="have plone-4"
            />
```

https://travis-ci.org/collective/collective.solr/jobs/156449782

**Comment by tomgross** [comment]
As it looks all the failing tests are related to the switch of the search template and its different DOM.

**Comment by tisto** [comment]
Todo list discussed with Tom via phone:

[ ] create a todo list for all doctests that need to be re-written as robot tests
[ ] remove all current doctests -> plone 4 build should be green then
[ ] implement the most important tests from todo list
[ ] enable robot tests to run against Plone 4

**Comment by tomgross** [comment]
@tisto All *unit*tests are now green on both versions **but** 2 robot tests fail on Plone 4.3. Do you have any hint to fix them?

**Comment by tomgross** [comment]
This is now ready to merge. What is done here:
- Remove skins folder and search profile
- Make boost script config available in control panel
- Fix all doctests on Plone 4 & 5
- AND: Fix a nasty bug, where Collections showed empty results, if Solr was queried

**Comment by tomgross** [comment]
@tisto @gforcada Can you please review and merge. I want to get out a Plone 5 compatible release as soon as possible to have a stable basis for my Solr training @ Plone conference in Boston.

File: plone-4.3.x.cfg [hunk 1]
```diff
@@ -1,2 +1,2 @@
 [buildout]
-extends = http://dist.plone.org/release/4.3.9/versions.cfg
+extends = http://dist.plone.org/release/4.3.11-pending/versions.cfg
```
Comment (gforcada) [inline plone-4.3.x.cfg]: Should we stick to released versions? In some days this link will no longer work, just like extends = http://dist.plone.org/release/4.3.10-pending/versions.cfg does not work... OTOH if there are some bits needed, then let's keep it but add an issue to update the link as soon as Eric does the final release

File: src/collective/solr/TODO.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Create robot test for
+
+ - wildcard search
```
Comment (gforcada) [inline src/collective/solr/TODO.rst]: is that still a TODO or should it be removed?

File: plone-4.3.x.cfg (hunk 1 above)
Comment (tomgross) [inline plone-4.3.x.cfg]: This is a leftover of an experiment. I reverted it to Plone 4.3.10

File: plone-5.0.x.cfg [hunk 3]
```diff
@@ -1,2 +1,2 @@
 [buildout]
-extends = http://dist.plone.org/release/5.0.5/versions.cfg
+extends = http://dist.plone.org/release/5.0.6-pending/versions.cfg
```
Comment (gforcada) [inline plone-5.0.x.cfg]: ditto :-)

File: src/collective/solr/TODO.rst (hunk 2 above)
Comment (tomgross) [inline src/collective/solr/TODO.rst]: It is. I agreed with @tisto to keep a list of doctest which are not ported to robot tests and this is the only one. I move it over to the robot files itself. It is more visible there I guess.

**Comment by tisto** [comment]
@tomgross I'm done with my review. I think overall this pull request is fine. The PAC problem is really really ugly and I wish we could solve that but I already run into a wall myself with that.

Maybe @gforcada can have a second look...

**Comment by tomgross** [comment]
@gforcada any objections to merge this?

**Comment by tisto** [comment]
@tomgross did you see my review comments? Can we please at least fix the robot test skip? Can you elaborate why the tests do not work on Plone 4?

**Comment by tomgross** [comment]
@tisto it works in real life but not in the tests. robot does not "see" the
facet dialog
Timo Stollenwerk <notifications@github.com> schrieb am Fr., 23. Sep. 2016
um 09:16:

**Comment by tomgross** [comment]
i would need help to fix these but I don't see this as a blocker for a
merge and alpha release.
Tom Gross <itconsense@gmail.com> schrieb am Fr., 23. Sep. 2016 um 09:19:

**Comment by tisto** [comment]
@tomgross did you try to debug into the robot test to see why it does not see the facet dialog?

**Comment by gforcada** [comment]
@tomgross I really don't have much time for a review, but at least yesterday night I cleaned up the branch by merging commits together and rebasing on top of master. Should I force push it? In general I would very much like to keep a clean history so that later on our future selves can understand what was going on

**Comment by tomgross** [comment]
@tisto i did but could not find anything. i did some experiments with
z-index but no success.
Sorry for beeing pushy here. I think one reason the adoption of Plone 5 is
so slow is the lack of released addons and I want to fix this AND I need a
released version of c.solr for my training in boston.
Gil Forcada Codinachs <notifications@github.com> schrieb am Fr., 23. Sep.
2016 um 09:27:

**Comment by tomgross** [comment]
@gforcada go ahead with cleaning up the history
Tom Gross <itconsense@gmail.com> schrieb am Fr., 23. Sep. 2016 um 09:40:

**Comment by tisto** [comment]
@tomgross I removed your Plone 4 robot test skips and it turns out the failures you skipped are real regressions in Plone 4:

1) The filter div is placed behind the Plone footer (I already fixed that)
2) If the user deselects a checkbox on the filter, the filter div closes and it is impossible for the user (and the robot test) to unselect a second checkbox.

I'm currently on vacation, so I don't have much time to look into that, but we can not skip tests that report real regressions...

**Comment by tisto** [comment]
3) Scenario: As anonymous user I can filter the test results by creation date: even though there is a filter on creation date, the search result returns two results instead of one (one should have been filtered). Again, this is a real regression in Plone 4.

**Comment by tisto** [comment]
Regarding 2)

I can fix the test with this test workaround:

```
I filter the search by portal type '${portal_type}'
  Click Button  xpath=//button[@id='search-filter-toggle']
  Wait until element is visible  xpath=//input[@id='query-portaltype-Collection']
  Unselect Checkbox  xpath=//input[@id='query-portaltype-Collection']
  # TEMP FIX START !!!
  # The search filter div closes when the user unselects a checkbox, this is a
  # regression that needs to be fixed.
  Click Button  xpath=//button[@id='search-filter-toggle']
  Wait until element is visible  xpath=//input[@id='query-portaltype-Collection']
  # TEMP FIX END !!!
  Unselect Checkbox  xpath=//input[@id='query-portaltype-Document']
```

We have to fix this properly of course.

**Comment by tisto** [comment]
@tomgross if you need help with the react setup @sneridagh can help you.

**Comment by tomgross** [comment]
@tisto Digging into ReactJS is on my list but not right now. I'm too busy with other stuff. If the release does not happen soon because nobody fixes the tests I'll release my own version of collective.solr and put it on a separate egg server. I see this as the lesser evil than working on an unmerged branch on github.

**Comment by tisto** [comment]
@tomgross would a release early next week be good enough for you? I can ask @sneridagh to look into the react specific parts. If we fix the regressions and if we have the acceptance tests pass for Plone4 I'm more than happy to make an alpha release.

**Comment by tomgross** [comment]
I guess we agreed this is OK except the two robot tests failing on Plone 4. I'll merge this now and continue in a new branch / PR. This PR is overloaded already.

**Comment by tomgross** [comment]
To be continued here: #144

**Comment by sneridagh** [comment]
@tomgross @tisto I will find time to take a look at the failing tests in Plone 4. Following it on #144.

**Pull request merged**